sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -152,6 +153,18 @@ case class Sort(
     global: Boolean,
     child: LogicalPlan) extends UnaryNode {
   override def output = child.output
+
+  override def resolveChildren(name: String, resolver: Resolver) = {
+    val input = child match {
+      case Project(list, c) => list.filter {
+        case Alias(g: GetField, _) => false
+        case Alias(g: GetItem, _) => false
```

Comment:
Can we just filter out everything except `AttributeReference`? I don't know all corner cases of `ORDER BY` and feel this way is safer.